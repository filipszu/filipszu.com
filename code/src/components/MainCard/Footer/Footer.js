import classes from "./Footer.module.css";

const footer = (props) => {
    return (
        <footer className={classes.Footer}>
            |&nbsp;<a target="_blank" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
            |&nbsp;<a target="_blank" href="https://github.com/filipszu">GitHub</a>&nbsp;
            |&nbsp;<a target="_blank" href="http://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
        </footer>
    );
};

export default footer;