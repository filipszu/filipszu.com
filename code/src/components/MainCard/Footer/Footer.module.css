.Footer{
	text-align: center;
	position: absolute;
	bottom: 0px;
	font-size: 32px;
	margin-bottom: 10px;
	width: 90%;	
}

.Footer a{
	text-decoration: none;
	border-bottom: 1px solid #A2EF00;
	color: #A2EF00; 	
}

.Footer a:hover{
	text-decoration: none;
	border-bottom: 1px solid #CB0077;
	color: #CB0077; 	
}