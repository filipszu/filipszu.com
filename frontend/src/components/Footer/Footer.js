import classes from "./Footer.module.css";
import { BrowserView, MobileView } from 'react-device-detect';

const footer = (props) => {
    return (
        <footer className={classes.Footer}>
            <BrowserView>
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/filipszu">GitHub</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://stackoverflow.com/users/9680496/filip-szulczewski">StackOverflow</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
            </BrowserView>
            <MobileView>
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/filipszu">GitHub</a>&nbsp;|
                <br />
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://stackoverflow.com/users/9680496/filip-szulczewski">StackOverflow</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
            </MobileView>
        </footer>
    );
};

export default footer;