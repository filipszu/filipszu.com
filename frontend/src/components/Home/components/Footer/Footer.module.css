.Footer{
	text-align: center;
	position: absolute;
	bottom: 0px;
	font-size: 32px;
	margin-bottom: 10px;
	width: 90%;	
}

.Footer a{
	text-decoration: none;
	border-bottom: 1px solid #A2EF00;
	color: #A2EF00; 	
}

.Footer a:hover{
	text-decoration: none;
	border-bottom: 1px solid #CB0077;
	color: #CB0077; 	
}

@media only screen and (max-width : 320px) {
	.Footer{
		font-size: 24px;
		line-height: 1;
	}
}

@media only screen and (max-width : 800px) {
	.Footer{
		font-size: 24px;
		line-height: 1;
	}
}