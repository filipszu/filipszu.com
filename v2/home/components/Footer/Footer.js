import classes from "./Footer.module.css";
import { isMobile, isTablet } from 'react-device-detect';

const footer = (props) => {
    let links = (
        <div>
            |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
            |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/filipszu">GitHub</a>&nbsp;
            |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://stackoverflow.com/users/9680496/filip-szulczewski">StackOverflow</a>&nbsp;
            |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
        </div>
    );

    if(isMobile || isTablet){
        links = (
            <div>
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://twitter.com/filipszu">@filipszu</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://github.com/filipszu">GitHub</a>&nbsp;|
                <br />
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://stackoverflow.com/users/9680496/filip-szulczewski">StackOverflow</a>&nbsp;
                |&nbsp;<a target="_blank" rel="nofollow noopener noreferrer" href="https://www.linkedin.com/in/filipszu">LinkedIn</a>&nbsp;|
            </div>
        );
    }

    return (
        <footer className={classes.Footer}>
            {links}
        </footer>
    );
};

export default footer;