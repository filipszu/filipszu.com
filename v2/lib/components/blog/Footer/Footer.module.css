.Footer{
    border-top: 2px dashed #CB0077;
    bottom: 0;
    border-radius: 0;
    background-color: rgba(0, 0, 0, .85);
    min-height: 20px;
    padding: 19px;
    margin-top: 1em;
    font-size: 1.3em;
    text-align: center;
}