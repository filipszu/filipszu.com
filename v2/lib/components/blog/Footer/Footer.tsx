import React from "react";
import classes from "./Footer.module.css";
import Link from "next/link";
export default function Footer() {
  const links = (
    <div>
      |&nbsp;<Link href="/">Home</Link>&nbsp; |&nbsp;
      <Link href="/blog">Blog</Link>&nbsp; |&nbsp;
      <a
        target="_blank"
        rel="nofollow noopener noreferrer"
        href="https://twitter.com/filipszu"
      >
        @filipszu
      </a>
      &nbsp; |&nbsp;
      <a
        target="_blank"
        rel="nofollow noopener noreferrer"
        href="https://github.com/filipszu"
      >
        GitHub
      </a>
      &nbsp; |&nbsp;
      <a
        target="_blank"
        rel="nofollow noopener noreferrer"
        href="https://stackoverflow.com/users/9680496/filip-szulczewski"
      >
        StackOverflow
      </a>
      &nbsp; |&nbsp;
      <a
        target="_blank"
        rel="nofollow noopener noreferrer"
        href="https://www.linkedin.com/in/filipszu"
      >
        LinkedIn
      </a>
      &nbsp;|
    </div>
  );

  return <div className={classes.Footer}>{links}</div>;
}
