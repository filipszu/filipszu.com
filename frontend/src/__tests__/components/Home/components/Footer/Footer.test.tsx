import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Footer, {
  TestID,
} from "../../../../../components/Home/components/Footer/Footer";

const hardcodedLinks = {
  twitter: { label: "@filipszu", url: "https://twitter.com/filipszu" },
  github: { label: "GitHub", url: "https://github.com/filipszu" },
  stackOverflow: {
    label: "StackOverflow",
    url: "https://stackoverflow.com/users/9680496/filip-szulczewski",
  },
  linkedIn: { label: "LinkedIn", url: "https://www.linkedin.com/in/filipszu" },
};

describe(`Footer component responsibble for rendering the links to other platforms.`, () => {
  test(`It will render correctly and match the snapshot.`, () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test(`[Given] it will render [Then] it will use the appropriate test-id: ${TestID}`, () => {
    render(<Footer />);
    Object.values(hardcodedLinks).forEach((link) => {
      const el = screen.getByTestId(TestID);
      expect(el).toBeInTheDocument();
    });
  });
  test(`[Given] the page will render [Then] there will be a link to all of the hardcoded websites with appropriate labels`, () => {
    render(<Footer />);
    Object.values(hardcodedLinks).forEach((link) => {
      const el = screen.getByText(link.label);
      expect(el).toBeInTheDocument();
    });
  });
  test(`[Given] the page will render [Then] there will be a link to all of the hardcoded websites with appropriate urls`, () => {
    render(<Footer />);

    Object.values(hardcodedLinks).forEach((link) => {
      expect(screen.getByRole("link", { name: link.label })).toHaveAttribute(
        "href",
        link.url
      );
    });
  });
});
