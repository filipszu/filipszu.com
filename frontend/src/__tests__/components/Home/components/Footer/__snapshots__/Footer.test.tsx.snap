// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component responsibble for rendering the links to other platforms. It will render correctly and match the snapshot. 1`] = `
<footer
  className="Footer"
>
  <div
    data-testid="szu-footer"
  >
    | 
    <a
      href="https://twitter.com/filipszu"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      @filipszu
    </a>
      | 
    <a
      href="https://github.com/filipszu"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      GitHub
    </a>
      | 
    <a
      href="https://stackoverflow.com/users/9680496/filip-szulczewski"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      StackOverflow
    </a>
      | 
    <a
      href="https://www.linkedin.com/in/filipszu"
      rel="nofollow noopener noreferrer"
      target="_blank"
    >
      LinkedIn
    </a>
     |
  </div>
</footer>
`;
